Lib/logging/__init__.py

```diff
@@ -800,17 +800,14 @@ def filter(self, record):
         .. versionchanged:: 3.2
 
            Allow filters to be just callables.
+
+        .. versionchanged: 3.9
+
+           Improve readability using efficient Python 3 builtins.
         """
-        rv = True
-        for f in self.filters:
-            if hasattr(f, 'filter'):
-                result = f.filter(record)
-            else:
-                result = f(record) # assume callable - will raise if not
-            if not result:
-                rv = False
-                break
-        return rv
+        filters = (getattr(f, 'filter', f) for f in self.filters)
+        return all(f(record) for f in filters) # assume callable - will raise if not
```

Comment:
This PR modifies behavior.
Before the execution was stopped on the first filter that returned a negative result.
After the change all filters are called.

Reply:
I don't believe it changes the behavior as you say. See https://docs.python.org/3/library/functions.html#all

I could add a unit test that specifies the short-cutting behavior, if that helps.

Reply:
I don't believe it changes the behavior as you say. See https://docs.python.org/3/library/functions.html#all

I could add a unit test that specifies the short-cutting behavior, if that helps.

Reply:
That's not true; `all()` exits at the first falsey value.

Reply:
Sorry, you are right.

Reply:
Honestly, I haven't found that the new implementation is better readable but ok if @zware like it.